Sources/ZcashLightClientKit/Block/Actions/RewindAction.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  RewindAction.swift
+//  
+//
+//  Created by Lukáš Korba on 09.08.2023.
+//
+
+import Foundation
+
+final class RewindAction {
+    let downloader: BlockDownloader
+    let rustBackend: ZcashRustBackendWelding
+    let downloaderService: BlockDownloaderService
+    let logger: Logger
+
+    init(container: DIContainer) {
+        downloader = container.resolve(BlockDownloader.self)
+        rustBackend = container.resolve(ZcashRustBackendWelding.self)
+        downloaderService = container.resolve(BlockDownloaderService.self)
+        logger = container.resolve(Logger.self)
+    }
+    
+    private func update(context: ActionContext) async -> ActionContext {
+        await context.update(state: .download)
```

Comment:
Rewinding can invalidate / change scan ranges, so this needs to be:
```suggestion
        await context.update(state: .processSuggestedScanRanges)
```

Reply:
Or if `.clearCache` is what everything needs to go through to get there:
```suggestion
        await context.update(state: .clearCache)
```